src/main/policy.c

```diff
@@ -80,30 +96,35 @@
         }                                                                      \
     }
 
-#define POLICY_SET_EXPRESSIONS_BASE_FIELD()                                    \
+#define POLICY_SET_EXPRESSIONS_FIELD()                                         \
     {                                                                          \
         if (exp_list) {                                                        \
+            PyObject *py_field_name = PyUnicode_FromString("expressions");     \
+            if (py_field_name == NULL) {                                       \
+                PyErr_Clear();                                                 \
+                return as_error_update(                                        \
+                    err, AEROSPIKE_ERR_CLIENT,                                 \
+                    "Unable to create Python unicode object");                 \
+            }                                                                  \
             PyObject *py_exp_list =                                            \
-                PyDict_GetItemString(py_policy, "expressions");                \
+                PyDict_GetItemWithError(py_policy, py_field_name);             \
+            if (py_exp_list == NULL && PyErr_Occurred()) {                     \
+                PyErr_Clear();                                                 \
+                Py_DECREF(py_field_name);                                      \
+                return as_error_update(err, AEROSPIKE_ERR_CLIENT,              \
+                                       "Unable to fetch expressions field "    \
+                                       "from policy dictionary");              \
+            }                                                                  \
+            Py_DECREF(py_field_name);                                          \
```

Comment:
Same as previous comment, couldn't you move `Py_DECREF(py_field_name);` before the if?

Reply:
See my comment above

Reply:
seen, lgtm